### PR TITLE
Optionally show namespaces of candidates in popup menu

### PR DESCRIPTION
**Pull request by xjeffw**
This matches the menu from the cider company-mode backend.
[screenshot](http://i.imgur.com/MbT6sAC.png)

### CODE REVIEW

**Comment by bbatsov** [comment]
Why is everything a `v`? In company you'd get things like `f` (function), `m` (macro) and so on.

Otherwise the change looks useful to me.

File: ac-cider.el [hunk 1]
```diff
@@ -68,11 +68,32 @@
...
 (defvar ac-cider-documentation-cache '())
 
+(defcustom ac-cider-show-ns t
+  "Non-nil means rows in the AC popup menu will show the namespace of the item."
+  :type 'boolean
+  :group 'ac-cider)
+
+(defun ac-cider-prepare-candidates (candidates)
+  "Apply changes for auto-complete to the candidates returned from cider-complete."
+  (cond
```
Comment (bbatsov) [inline ac-cider.el]: I'd use `if` here.

File: ac-cider.el [hunk 2]
```diff
@@ -68,11 +68,32 @@
...
+     candidates))
+   (t
+    candidates)))
+
 (defun ac-cider-candidates-everything ()
   "Return all candidates for a symbol at point."
   (setq ac-cider-documentation-cache nil)
   (when (cider-connected-p)
-    (cider-complete ac-prefix)))
+    (let ((candidates (cider-complete ac-prefix)))
```
Comment (bbatsov) [inline ac-cider.el]: This `let` seems a bit redundant to me.

**Comment by bbatsov** [comment]
I'd also mention this change in the readme.

**Comment by alexander-yakushev** [comment]
@bbatsov:

> Why is everything a v? In company you'd get things like f (function), m (macro) and so on.

Because of AC's stupid system that only allows different one-letter tags if they come from different completion sources. In other words, to get `n`, `f`, and `v` AC has to query the completion backend **three** times.

**Comment by bbatsov** [comment]
Well, that sucks. Luckily I'm a happy company user. :-)

**Comment by alexander-yakushev** [comment]
After using company-mode for a few months now I'd say that it has its sucky moments as well :).

**Comment by alexander-yakushev** [comment]
Great, thanks! Just splice everything into a single commit and we're good to merge :).

**Comment by jeffwk** [comment]
Okay, done.

**Pull request merged**